deploy/crds/metal3.io_baremetalhosts_crd.yaml

```diff
@@ -172,6 +172,61 @@ spec:
             online:
               description: Should the server be online?
               type: boolean
+            raid:
+              description: RAID configuration for bare metal server
+              properties:
+                cleanFlag:
+                  description: if cleanFlag is true will remove old configuration
+                    and set new configuration
+                  type: boolean
+                volumes:
+                  description: The list of logical disks
+                  items:
+                    description: RAIDVolume defines the desired configuration of volume
+                      in RAID
+                    properties:
```

Comment:
We need a way to provide `controller=software`. Maybe simply a boolean flag `isSoftwareRAID`?

Reply:
@dtantsur we already provide a way for the BMC driver to [specify what RAID controller to use](https://github.com/metal3-io/baremetal-operator/blob/master/pkg/bmc/access.go#L52) - can we not just infer from that being set to "agent" that we need to set `controller=software`?
i.e. is there are reason to allow users to override the driver to use software RAID when a HW implementation is available? I'd really like to reduce the complexity of the interface and the opportunities for the user to supply invalid config.